crates/sui-types/src/messages.rs

```diff
@@ -2081,214 +2081,3 @@ pub enum QuorumDriverResponse {
     // TODO: Change to CertifiedTransactionEffects eventually.
     EffectsCert(Box<(CertifiedTransaction, CertifiedTransactionEffects)>),
 }
-
-// Epoch related data structures.
-
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct EpochInfo {
-    /// The committee of this epoch.
-    committee: Committee,
-    /// The first checkpoint included in this epoch.
-    first_checkpoint: CheckpointSequenceNumber,
-    /// Digest of the epoch info from the previous epoch.
-    prev_digest: EpochInfoDigest,
-}
-
-impl EpochInfo {
-    pub fn new(
-        committee: Committee,
-        first_checkpoint: CheckpointSequenceNumber,
-        prev_digest: EpochInfoDigest,
-    ) -> Self {
-        Self {
-            committee,
-            first_checkpoint,
-            prev_digest,
-        }
-    }
-
-    pub fn epoch(&self) -> EpochId {
-        self.committee.epoch
-    }
-
-    pub fn committee(&self) -> &Committee {
-        &self.committee
-    }
-
-    pub fn into_committee(self) -> Committee {
-        self.committee
-    }
-
-    pub fn first_checkpoint(&self) -> &CheckpointSequenceNumber {
-        &self.first_checkpoint
-    }
-
-    pub fn prev_epoch_info_digest(&self) -> &EpochInfoDigest {
-        &self.prev_digest
-    }
-
-    pub fn digest(&self) -> EpochInfoDigest {
-        sha3_hash(self)
-    }
-}
-
-pub type EpochInfoDigest = [u8; 32];
-
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct EpochEnvelop<S> {
-    pub epoch_info: EpochInfo,
-    pub auth_sign_info: S,
-}
-
-pub type GenesisEpoch = EpochEnvelop<EmptySignInfo>;
```

Comment:
it's removed but not sure if worth asking: don't have we validators sigs in in genesis.blob to construct a certified `GenesisEpoch`?

Reply:
The GenesisEpoch today is not certified. But that's a good question. I am not quite sure what we plan to do with the genesis committee